pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -1130,28 +1155,47 @@ func (m *kubeGenericRuntimeManager) SyncPodContainer(pod *v1.Pod, podStatus *kub
 					return err
 				}
 			}
-			if newPodCgLimit < currPodCgLimit {
+			if newPodCgValue < currPodCgValue {
 				err = setPodCgroupConfig(rName)
 			}
 			return err
 		}
-		if len(podContainerChanges.ContainersToUpdate[v1.ResourceMemory]) > 0 || len(podContainerChanges.ContainersToRestart) > 0 {
+		if len(podContainerChanges.ContainersToUpdate["memory-limit"]) > 0 || len(podContainerChanges.ContainersToRestart) > 0 {
 			currentPodMemoryLimit, err := pcm.GetPodCgroupMemoryConfig(pod)
 			if err != nil {
 				klog.Errorf("GetPodCgroupMemoryConfig for pod %s failed: %v", pod.Name, err)
 				return
 			}
-			if errUpdate := resizeContainers(v1.ResourceMemory, int64(currentPodMemoryLimit), *podResources.Memory); errUpdate != nil {
+			currentPodMemoryUsage, err := pcm.GetPodCgroupMemoryUsage(pod)
+			if err != nil {
+				klog.Errorf("GetPodCgroupMemoryUsage for pod %s failed: %v", pod.Name, err)
+				return
+			}
+			if currentPodMemoryUsage >= uint64(*podResources.Memory) {
```

Comment:
if this case is hit. surely a error case. but how could VPA get this error and act on it ?

Reply:
I think this can happen with memory-backed emptyDir that VPA may not be aware of.